### ISSUE TITLE
rpool does not open on boot

### ISSUE THREAD

**Issue by mikeplem** [zfsonlinux/zfs#9618]
### System information

Type | Version/Name
 --- | --- 
Distribution Name	| Debian
Distribution Version	| 10
Linux Kernel	| 4.19.67-2+deb10u1
Architecture	| x86_64
ZFS Version	| 0.8.2-2~bpo10+1
SPL Version	| 0.8.2-2~bpo10+1

### Describe the problem you're observing
The Debian Buster Root on ZFS documentation is missing a GRUB_CMDLINE_LINUX_DEFAULT grub argument to allow rpool to open on boot. I configured a native ZFS encrypted system and I would not be given the decryption password prompt until I updated GRUB_CMDLINE_LINUX_DEFAULT to include **boot=zfs $bootfs**

I found the solution from this post
<https://askubuntu.com/questions/898466/ubuntu-16-04-2-lts-on-zfs-root-unable-to-boot>

### Describe how to reproduce the problem
Remove the boot=zfs $bootfs arguments from GRUB_CMDLINE_LINUX_DEFAULT

### Include any warning/errors/backtraces from the system logs

No errors were shown. The boot process appeared to stop before the ZFS decryption prompt was supposed to show.

### My Running /etc/default/grub

```
# If you change this file, run 'update-grub' afterwards to update
# /boot/grub/grub.cfg.
# For full documentation of the options in this file, see:
#   info -f grub -n 'Simple configuration'

GRUB_DEFAULT=0
GRUB_TIMEOUT=5
GRUB_DISTRIBUTOR=`lsb_release -i -s 2> /dev/null || echo Debian`
GRUB_CMDLINE_LINUX_DEFAULT="boot=zfs $bootfs quiet"
#GRUB_CMDLINE_LINUX=""
GRUB_CMDLINE_LINUX="root=ZFS=rpool/ROOT/debian"

# Uncomment to enable BadRAM filtering, modify to suit your needs
# This works with Linux (no patch required) and with any kernel that obtains
# the memory map information from GRUB (GNU Mach, kernel of FreeBSD ...)
#GRUB_BADRAM="0x01234567,0xfefefefe,0x89abcdef,0xefefefef"

# Uncomment to disable graphical terminal (grub-pc only)
#GRUB_TERMINAL=console

# The resolution used on graphical terminal
# note that you can use only modes which your graphic card supports via VBE
# you can see them in real GRUB with the command `vbeinfo'
#GRUB_GFXMODE=640x480

# Uncomment if you don't want GRUB to pass "root=UUID=xxx" parameter to Linux
#GRUB_DISABLE_LINUX_UUID=true

# Uncomment to disable generation of recovery mode menu entries
#GRUB_DISABLE_RECOVERY="true"

# Uncomment to get a beep at grub start
#GRUB_INIT_TUNE="480 440 1"
```

### Patch to grub
```
diff --git a/Debian-Buster-Root-on-ZFS.md b/Debian-Buster-Root-on-ZFS.md
index d9eea21..dd04ddb 100644
--- a/Debian-Buster-Root-on-ZFS.md
+++ b/Debian-Buster-Root-on-ZFS.md
@@ -534,34 +534,41 @@ In the future, you will likely want to take snapshots before each upgrade, and r
     mount | grep -v zfs | tac | awk '/\/mnt/ {print $3}' | xargs -i{} umount -lf {}
     zpool export -a
 
-6.4  Reboot:
+6.4  Setup grub to open rpool on boot
+
+    vi /etc/default/grub
+    Add boot=zfs $bootfs to: GRUB_CMDLINE_LINUX_DEFAULT
+    GRUB_CMDLINE_LINUX_DEFAULT="boot=zfs $bootfs quiet"
+    Save and quit.
+
+6.5  Reboot:
 
     reboot
 
-6.5  Wait for the newly installed system to boot normally. Login as root.
+6.6  Wait for the newly installed system to boot normally. Login as root.
 
-6.6  Create a user account:
+6.7  Create a user account:
 
     zfs create rpool/home/YOURUSERNAME
     adduser YOURUSERNAME
     cp -a /etc/skel/. /home/YOURUSERNAME
     chown -R YOURUSERNAME:YOURUSERNAME /home/YOURUSERNAME
 
-6.7  Add your user account to the default set of groups for an administrator:
+6.8  Add your user account to the default set of groups for an administrator:
 
     usermod -a -G audio,cdrom,dip,floppy,netdev,plugdev,sudo,video YOURUSERNAME
 
-6.8   Mirror GRUB
+6.9   Mirror GRUB
 
 If you installed to multiple disks, install GRUB on the additional disks:
 
-6.8a  For legacy (BIOS) booting:
+6.9a  For legacy (BIOS) booting:
 
     dpkg-reconfigure grub-pc
     Hit enter until you get to the device selection screen.
     Select (using the space bar) all of the disks (not partitions) in your pool.
 
-6.8b  UEFI
+6.9b  UEFI
 
     umount /boot/efi
 ```

**Comment by Ornias1993** [comment]
Is this only the case with your encrypted pool, or normal pools too?

**Comment by mikeplem** [comment]
Aside from the boot pool (bpool) I only have the root pool (rpool).

**Comment by Ornias1993** [comment]
> No errors were shown. 
It seems you skipped 5.4 , if you do 5.4 you will get the appropiate errors to add to your Issue :)

A lot of people seem to be running without this flag without issue, so it might be worth while finding out while you and 2 people 1-2 years ago did require this flag.

That being said, I don't see why adding this would be a problem either.

**Comment by mikeplem** [comment]
I don't remember when I reverted step 5.4 but since I don't remember it, I probably did comment out the console line before I rebooted. I do know that I had enabled the console at some point but it must not have been when I really needed it.

I would agree that such a small number of people needing to add the arguments is interesting when no one else needs it. It is over my head but what does boot=zfs do under the covers?
